color_refinement_helper.py

```diff
@@ -195,7 +200,60 @@ def modules_to_graph(modules: ModularDecomposition) -> Graph:
         graph.add_vertex(vertex)
     else:
         graph = create_graph_helper(sorted(list(edges)))
-    return graph
+
+    mapping = {}
+    for vertex, label in label_mapping.items():
+        mapping[vertex] = graph.find_vertex(label)
+
+    return graph, mapping
+
+
+def determine_module_connectivity(md: ModularDecomposition):
+    connected_modules = []
+    disconnected_modules = []
+    for module in (module for module in md if len(module) > 1):
+        if module[1] in module[0].neighbours:
+            # Connected, i.e. serial
+            connected_modules.append(module)
+        else:
+            # Disconnected, i.e. parallel
+            disconnected_modules.append(module)
+
+    return connected_modules, disconnected_modules
+
+
+def find_isomorphic_modules(md: ModularDecomposition):
+    isomorphic_modules = {}
+    for module in md:
+        isomorphic_modules.setdefault(len(module), []).append(module)
+    return isomorphic_modules
+
+
+def modules_to_graph_with_module_isomorphism(md: ModularDecomposition):
+    graph, mapping = modules_to_graph(md)
+
+    connected_md, disconnected_md = determine_module_connectivity(md)
+    isomorphic_connected_modules = find_isomorphic_modules(connected_md)
```

Comment:
Je kunt hier ook `group_by(connected_md, group_rule=lambda m: len(m))` doen

Reply:
Neem ik mee in een andere branch.